mautic/mautic
# SAML SSO support

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | Y
| Related user documentation PR URL | todo
| Related developer documentation PR URL | todo
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

This adds SAML SSO support to Mautic.

#### Steps to test this PR:
1. Apply this PR and clear your cache
2. Register for a free IDP (identify provider) at http://www.ssocircle.com/en/
3. After creating the account and logging in, go to Manage Metadata.
4. Click Add a New Service Provider
5. Enter the URL of your Mautic for the FQDN
6. Check all three optional attributes
7. Visit http://yourmautic.com/saml/metadata.xml and copy the XML into the metadata box and submit.
8. Back on the "Manage Metadata" page, click on SSOCircle Public IDP Metadata link below the link to create a new service provider. It'll download a different XML page.
9. Login to Mautic and go to the configuration -> User/Authentication Settings
10. Upload SSOCircle xml file as the IDP metadata. For the attributes, fill in EmailAddress, FirstName, and LastName into their corresponding attributes fields. Select the default role and save.
11. Logout and browse to  http://yourmautic.com/saml/login.  You should be redirected to SSOCircle. Login and if all goes well, you should be redirected back to Mautic, a new user created (if email/username doesn't exist), and the user is now logged in with the defaulted role.
